analyses/infercnv-consensus-cell-type/template-notebooks/SCPCP000015_explore-infercnv-results.Rmd

```diff
@@ -0,0 +1,418 @@
+To help contextualize this image, here is the table of known CNVs in Ewing sarcoma used for validation:
+
+```{r}
+validation_df
+```
+
+
+![InferCNV heatmap](`r infercnv_png`)
+
+
+## Distribution of CNVs per cell
+
+This section contains figures showing the total number of CNVs detected per cell.
+
+### CNVs in reference vs query cells
+
+The figures below compare the number of CNVs per cell between reference and query cells.
+Ideally, we'll see a bimodal distribution where reference cells are at/near zero, and there is a peak of higher values which may represent tumor cells.
+In the density plots below, distributions are colored based on whether the cells are in the reference or not (aka, "unknown" cells).
+
+```{r}
+# First, construct data frame of total `has_cnv` counts, with other variables needed for plots
+has_cnv_df <- infercnv_df |>
+  dplyr::select(
+    barcodes,
+    starts_with("has_cnv_")
+  ) |>
+  tidyr::pivot_longer(
+    starts_with("has_cnv_"),
+    names_to = "chr",
+    values_to = "cnv"
+  ) |>
+  # sum all values for each
+  dplyr::group_by(barcodes) |>
+  dplyr::summarize(total_cnv_per_cell = sum(cnv)) |>
+  dplyr::ungroup() |>
+  # bring back metadata
+  dplyr::inner_join(
+    infercnv_metadata_df,
+    by = "barcodes"
+  ) |>
+  # join in SCE information
+  dplyr::left_join(umap_df)
+```
+
+
+```{r fig.width = 10, fig.height = 5}
+ggplot(has_cnv_df) +
+  aes(x = total_cnv_per_cell, fill = cell_group) +
+  geom_density(alpha = 0.5) +
+  theme(legend.position = "bottom")
+```
+
+### CNVs across cell types
+
+Next, we'll look at the per-cell distributions of CNVs across cell types, using labels derived from the `cell-type-ewings` module as follows:
+
+* `tumor-endo`: Cells labeled as tumor by cell-type-ewings, but endothelial by `cell-type-consensus`
+* `tumor-immune`: Cells labeled as tumor by cell-type-ewings, but immune by `cell-type-consensus`
+* `tumor`: Cells labeled as tumor by `cell-type-ewings`
+* `endo`: An endothelial cell type
+* `immune`: An immune cell type
+* If none of the above, their existing `cell-type-ewings` annotation
+
+Ideally, we'll see that tumor and Unknown cells from will have relatively higher CNV scores compared to other cell types, in particular those cell types in the reference.
+
+```{r fig.height=12, message=FALSE, warning=FALSE}
+# drop cells from other libraries
+query_has_cnv_df <- has_cnv_df |>
+  dplyr::filter(library_id == params$library_id)
+
+ggplot(query_has_cnv_df) +
+  aes(
+    y = celltype,
+    x = total_cnv_per_cell,
+    fill = cell_group
+  ) +
+  ggridges::geom_density_ridges2(alpha = 0.7, scale = 0.85)
+```
+
+
+## UMAPs
+
+Next, we'll visualize the total CNV counts as a UMAP.
+Below, the UMAP colored by cell type is shown on the left, and the UMAP colored by total CNV counts is shown on the right.
+Ideally, unknown and tumor cells will overlap with regions in the UMAP with higher total CNV.
+
+
+```{r, fig.height = 10}
+celltype_umap <- ggplot(umap_df) +
```

Comment:
Here's where I think it would be more beneficial to just plot our three groups as mentioned in my previous comment. I like have the UMAP looking at total CNV because generally you do see that the cells we are labeling as tumor have the highest CNV. At least that's what this sample looks like, who knows what the rest of them look like.